src/alluvium/avro.py

```diff
@@ -0,0 +1,251 @@
+"""
+Defines the schema you expect from your CDC events.
+
+Provides synctactic sugar for Any types and Either Types.
+"""
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+
+class InvalidSchemaException(Exception):
+    def __init__(self, field_type, message, name=None):
+        self.error_field_type = field_type
+        self.name = name
+        super(InvalidSchemaException, self).__init__(
+            self, "Error with field <{} {}>. {}".format(field_type, name, message)
+        )
+
+
+class AvroPrimitiveType(Enum):
+    NULL = "null"
+    BOOLEAN = "boolean"
+    INTEGER = "int"
+    LONG = "long"
+    FLOAT = "float"
+    DOUBLE = "double"
+    BYTES = "bytes"
+    STRING = "string"
+
+
+class AvroComplexType(Enum):
+    RECORD = "record"
+    ENUM = "enum"
+    ARRAY = "array"
+    MAP = "map"
+    UNIONS = "unions"
+    FIXED = "fixed"
+
+
+class AvroCustomType(Enum):
+    ANY = "any"
+
+
+def resolve_primitive_type(avro_type: Union[AvroPrimitiveType, AvroCustomType]):
+    if isinstance(avro_type, AvroPrimitiveType):
+        return avro_type.value
+    if avro_type == AvroCustomType.ANY:
+        return AvroUnionSchema([primitive for primitive in AvroPrimitiveType]).avro_schema()
+    raise ValueError("Invalid base type provided")
+
+
+class FieldOrdering(Enum):
+    ASCENDING = "ascending"
+    DESCENDING = "descending"
+    IGNORE = "ignore"
+
+
+class AvroSchema(ABC):
+    def __init__(self, avro_type: Union[AvroPrimitiveType, AvroComplexType]):
+        self.type = avro_type
+
+    @abstractmethod
+    def avro_schema(self):
```

Comment:
Why was avro_schema(self) added to the AvroSchema abstract class?

Reply:
This function is expected to be implemented by the schemas that inherit this abstract class. The main point of utility is to convert a python object representation of avro schemas into a json representation.